### ISSUE TITLE
Floccus 4.1.0 duplicates bookmarks on the server

### ISSUE THREAD

**Issue by Newbear-dot** [marcelklehr/floccus#544]
**Describe the bug**
It seems that like described in #382 v4.1.0 still duplicates folders and bookmarks when running into a sync error.

**To Reproduce**
I have two instances of floccus, one in Firefox 76.0.1 on MacOS 10.15.5, one in Firefox 76.0.1 on FreeBSD. I have a Nextcloud 18.0.4 instance with latest Bookmark app. Floccus syncs fine (initial sync) on MacOS but most of the times on the first sync on FreeBSD runs into "Failed PATCH request" (see log). It seems that as soon as that happens, Floccus starts to create duplicate entries in Firefox and on the server.
Starting all over again (i.e. deleting all bookmarks on Nextcloud via web interface, importing them again and rerunning the syncs) recreates this behaviour.

**Desktop (please complete the following information):**
 - OS: MacOS 10.15.5, FreBSD 12.1
 - Browser: Firefox
 - Browser Version: 76.0.1 (in both cases)
 - Floccus version: 4.0.1
 - Floccus sync method: Nextcloud

**Server (please complete the following information):**
 - OS: FreeBSD 11.3
 - Nextcloud version: 18.0.4
 - Bookmarks app version: 3.0.13

Note: Please write down the actual version numbers instead of writing 'latest'.

**Debug log**
It is often useful to provide a debug log file along with the issue. You can obtain a debug log of the most recent sync runs by clicking 'Debug log' in the bottom right corner of the floccus panel in your browser.

Note: As your debug log usually contains large parts of your bookmark data, instead of attaching it publicly to the issue here, I recommend uploading it privately at the following URL https://klehranlage.net/index.php/s/7nn82NSBsfQKdsJ

- [X] Debug log provided

**Additional context**

**Comment by marcelklehr** [comment]
Hello @Newbear-dot 
I'm sorry for the trouble and will try to troubleshoot with your help. Thank you for taking the time to diligently fill out the issue template.

My initial diagnosis would be that the logs show missing cache entries. These should be there if the run in the log is the second run. It may be worthwhile to try to remove the floccus account entirely and recreate it, if you haven't tried this yet, to initialize a new cache entry in the extension storage.

Also, I assume we are talking about v4.1.0 as stated in the title as opposed to v4.0.1 as stated in the text body?

**Comment by Newbear-dot** [comment]
Hello @marcelklehr
 
first of all thank *you* for creating, maintaining and supporting this great piece of software.

Indeed I am talking about v4.1.0, sorry for that. 

Also, I think you are right, the duplicates always seem to have appeared after an error (e.g. status 400, which probably causes floccus to restart syncs?). 
Roughly my approach was: I had duplicates, so I cleared out the Nextcloud bookmarks completely via the web interface, then rerun the sync, then the FF bookmarks where empty, then I imported bookmarks again via Nextcloud web interface, then rerun the sync and then, as soon as something went wrong, it seems like the duplicates started to appear (could have been during the first sync, could have been later).

I am not sure if I understand you correctly regarding initializing a new cache entry, but I removed the floccus extension directory completely from the FF profile and reinstalled the extension, afterwards I restarted the sync and so far it works.

Nonetheless, I assume that if the sync errors reappear (which are a different story altogether of course, I am not sure whether the sporadic http 400 errors have something to do with my Nextcloud and/or Apache config), also the duplicates will reappear?
